kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -0,0 +1,770 @@
+        let tcx = self.tcx();
+        let span = tcx.def_span(def_id);
+
+        // We have to be a bit careful when retrieving names from def ids. For instance,
+        // due to reexports, [`TyCtxt::def_path_str`](TyCtxt::def_path_str) might give
+        // different names depending on the def id on which it is called, even though
+        // those def ids might actually identify the same definition.
+        // For instance: `std::boxed::Box` and `alloc::boxed::Box` are actually
+        // the same (the first one is a reexport).
+        // This is why we implement a custom function to retrieve the original name
+        // (though this makes us lose aliases - we may want to investigate this
+        // issue in the future).
+
+        // We lookup the path associated to an id, and convert it to a name.
+        // Paths very precisely identify where an item is. There are important
+        // subcases, like the items in an `Impl` block:
+        // ```
+        // impl<T> List<T> {
+        //   fn new() ...
+        // }
+        // ```
+        //
+        // One issue here is that "List" *doesn't appear* in the path, which would
+        // look like the following:
+        //
+        //   `TypeNS("Crate") :: Impl :: ValueNs("new")`
+        //                       ^^^
+        //           This is where "List" should be
+        //
+        // For this reason, whenever we find an `Impl` path element, we actually
+        // lookup the type of the sub-path, from which we can derive a name.
+        //
+        // Besides, as there may be several "impl" blocks for one type, each impl
+        // block is identified by a unique number (rustc calls this a
+        // "disambiguator"), which we grab.
+        //
+        // Example:
+        // ========
+        // For instance, if we write the following code in crate `test` and module
+        // `bla`:
+        // ```
+        // impl<T> Foo<T> {
+        //   fn foo() { ... }
+        // }
+        //
+        // impl<T> Foo<T> {
+        //   fn bar() { ... }
+        // }
+        // ```
+        //
+        // The names we will generate for `foo` and `bar` are:
+        // `[Ident("test"), Ident("bla"), Ident("Foo"), Disambiguator(0), Ident("foo")]`
+        // `[Ident("test"), Ident("bla"), Ident("Foo"), Disambiguator(1), Ident("bar")]`
+        let mut found_crate_name = false;
+        let mut name: Vec<PathElem> = Vec::new();
+
+        let def_path = tcx.def_path(def_id);
+        let crate_name = tcx.crate_name(def_path.krate).to_string();
+
+        let parents: Vec<_> = {
+            let mut parents = vec![def_id];
+            let mut cur_id = def_id;
+            while let Some(parent) = tcx.opt_parent(cur_id) {
+                parents.push(parent);
+                cur_id = parent;
+            }
+            parents.into_iter().rev().collect()
+        };
+
+        // Rk.: below we try to be as tight as possible with regards to sanity
```

Comment:
That's a comment copied from Charon, "Rk" is short for @R1kM I believe

Reply:
I believe it stands for "Remark" here

Reply:
Hah that makes so much more sense x)

Reply:
Fun conversation :)

I was hesitant to remove the comments because we may need to import any future upstream changes to this function, so I wanted to minimize the diff. @tautschnig Let me know if you think the comments create confusion, and I'll remove them.